src/test/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.ComHost.Tests/ClsidMapTests.cs

```diff
@@ -0,0 +1,178 @@
+            JObject clsidMap = CreateClsidMap(sharedTestState.ComLibraryFixture);
+            JProperty comVisibleEntry = clsidMap.Property(SharedTestState.ComVisibleNonPublicNestedGuid);
+            Assert.Null(comVisibleEntry);
+        }
+
+        [Fact]
+        public void PublicComVisibleTypeWithDuplicateGuidThrows()
+        {
+            var exception = Assert.Throws<ConflictingGuidException>(() => CreateClsidMap(sharedTestState.ComLibraryConflictingGuidFixture));
+            Assert.Equal(Guid.Parse(SharedTestState.ConflictingGuid), exception.Guid);
+            Assert.Equal(SharedTestState.ConflictingGuidTypeName1, exception.TypeName1);
+            Assert.Equal(SharedTestState.ConflictingGuidTypeName2, exception.TypeName2);
+        }
+
+        [Fact]
+        public void DefaultProgIdIsTypeName()
+        {
+            JObject clsidMap = CreateClsidMap(sharedTestState.ComLibraryFixture);
+            JProperty comVisibleEntry = clsidMap.Property(SharedTestState.ComVisibleGuid);
+            Assert.NotNull(comVisibleEntry);
+            JObject entry = (JObject)comVisibleEntry.Value;
+            Assert.Equal(SharedTestState.ComVisibleTypeName, entry.Property("progid").Value.ToString());
+        }
+
+        [Fact]
+        public void ExplicitProgIdUsed()
+        {
+            JObject clsidMap = CreateClsidMap(sharedTestState.ComLibraryFixture);
+            JProperty comVisibleEntry = clsidMap.Property(SharedTestState.ComVisibleCustomProgIdGuid);
+            Assert.NotNull(comVisibleEntry);
+            JObject entry = (JObject)comVisibleEntry.Value;
+            Assert.Equal(SharedTestState.ComVisibleCustomProgIdProgId, entry.Property("progid").Value.ToString());
+        }
+
+        [Fact]
+        public void ExplicitlyEmptyProgIdNotInClsidMap()
+        {
+            JObject clsidMap = CreateClsidMap(sharedTestState.ComLibraryFixture);
+            JProperty comVisibleEntry = clsidMap.Property(SharedTestState.ExplicitNoProgIdGuid);
+            Assert.NotNull(comVisibleEntry);
+            JObject entry = (JObject)comVisibleEntry.Value;
+            Assert.Null(entry.Property("progid"));
+        }
+
+        private JObject CreateClsidMap(TestProjectFixture project)
+        {
+            using var testDirectory = TestDirectory.Create();
+            string clsidMapPath = Path.Combine(testDirectory.Path, "test.clsidmap");
+            using (var assemblyStream = new FileStream(project.TestProject.AppDll, FileMode.Open, FileAccess.Read, FileShare.Delete | FileShare.Read))
+            {
+                using (PEReader peReader = new PEReader(assemblyStream))
+                {
+                    if (peReader.HasMetadata)
+                    {
+                        MetadataReader reader = peReader.GetMetadataReader();
+                        ClsidMap.Create(reader, clsidMapPath);
+                    }
+                }
+            }
+
+            using var clsidMapFile = File.OpenText(clsidMapPath);
```

Comment:
Please use at least one level of brace scoping here. this is really terse for no real reason.

Reply:
Yeah the new VS code helpers for the `using statement` feature are really aggressive. I'll switch it back to the traditional style.

Reply:
Thanks!